gov/politeia/proposals.go

```diff
@@ -188,26 +207,61 @@ func (db *ProposalDB) saveProposals(URLParams string) (int, error) {
 
 		err = db.dbP.Save(val)
 
-		// In the rare case scenario that the current proposal has a duplicate refID
-		// append an integer value to it till it becomes unique.
+		// When a duplicate CensorshipRecord struct is detected this "already exists"
+		// error is thrown, this means that some edits were made to an older version
+		// of the proposal and its fixed by updating the new changes. In another
+		// case that is more rare, is that the current proposal could have a Name
+		// that generates a RefID similar to one already in the db and appending
+		// integers to it till it becomes unique is the solution.
 		if err == storm.ErrAlreadyExists {
-			c := 1
-			for {
-				val.RefID += strconv.Itoa(c)
-				err = db.dbP.Save(val)
-				if err == nil || err != storm.ErrAlreadyExists {
-					break
+			// Check if the proposal token already exists in the db.
+			data, newErr := db.proposal("TokenVal", val.TokenVal)
+			if newErr == nil && data != nil {
+				// The proposal token already exists thus trigger an update with
+				// the latest details.
+				valCopy := *val
+				valCopy.ID = data.ID
+				prefixStr := ""
+
+				for k := 1; k <= maxLoop; k++ {
+					valCopy.RefID += prefixStr
+					// Attempt to update the old entry.
+					newErr = db.dbP.Update(&valCopy)
+					if newErr == nil || newErr != storm.ErrAlreadyExists {
+						err = nil
+						break
+					}
+					prefixStr = strconv.Itoa(k)
+				}
+			}
+
+			// First try wasn't successful if newErr != nil.
+			if newErr != nil {
+
+				for c := 1; c <= maxLoop; c++ {
+					// Drop the previously assigned ID.
+					val.ID = 0
```

Comment:
`db.dbP.Save(val)` automatically assigns the `ID` value thus dropping the ID value ensures that a conflict is not likely to arise just in case another proposal was saved since the last time `db.dbP.Save(val)` run.